client/asset/eth/eth.go

```diff
@@ -2016,6 +2015,73 @@ func (w *TokenWallet) EstimateRegistrationTxFee(feeRate uint64) uint64 {
 		return math.MaxUint64
 	}
 	return g.Transfer * feeRate
+}
+
+// EstimateSendTxFee returns a tx fee estimate for sending provided amount using
+// the provided feeRate.
+func (w *ETHWallet) EstimateSendTxFee(value, _ uint64, subtract bool) (fee uint64, err error) {
+	if subtract {
+		return 0, fmt.Errorf("wallet does not support checking network fee for withdrawal")
```

Comment:
I wonder why that is ? there is no difference from wallet perspective whether its send or withdraw (regardless of `subtract` value fee will be the same),

and this method shouldn't be smart enough to differentiate between send/withdraw anyway (I think)

Reply:
It is an error to call this method with subtract `true`. Even though `core` performs an `isWithdrawer` check.

Reply:
- for TokenWallet Withdraw probably doesn't even make sense,
- for ETHWallet it probably does but maybe treating it same way as TokenWallet is better indeed

yeah, makes sense

Reply:
We don't know how to take fees from the amount...

Not possible for tokens at all because the fee is eth. We can't convert the token to eth on demand.

Not possible for eth because the fee is decided based on factors we can't control and can't always predict. We set the max we are willing to pay per gas and consensus decides the rest.

Anyway, I think correct to error here.